sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -417,6 +472,42 @@ protected internal virtual BlobClient WithClientSideEncryptionOptionsCore(Client
                 overwrite: false,
                 cancellationToken: cancellationToken);
 
+        /// <summary>
+        /// The <see cref="Upload(BinaryData, CancellationToken)"/> operation
+        /// creates a new block blob or throws if the blob already exists.
+        ///
+        /// For partial block blob updates and other advanced features, please
+        /// see <see cref="BlockBlobClient"/>.  To create or modify page or
+        /// append blobs, please see <see cref="PageBlobClient"/> or
+        /// <see cref="AppendBlobClient"/>.
+        ///
+        /// For more information, see
+        /// <see href="https://docs.microsoft.com/rest/api/storageservices/put-blob">
+        /// Put Blob</see>.
+        /// </summary>
+        /// <param name="content">
+        /// A <see cref="BinaryData"/> containing the content to upload.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// state of the updated block blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<BlobContentInfo> Upload(
+            BinaryData content,
+            CancellationToken cancellationToken) =>
+            Upload(
+                content,
+                overwrite: false,
```

Comment:
**I think this is the right call**, but we slightly bungled this in the other Upload methods so it's worth pausing to consider.  

If you start with `Upload(Stream)` it won't overwrite, but those semantics are lost as soon as you try to set `options` to anything.  If a customer updates existing code to add a `Content-Type`, they may inadvertently switch to overwriting.  We could decide to not copy that pattern, but then `Upload(Stream)` and `Upload(BinaryData)` are inconsistent which is awful for different reasons.  I'd personally rather stay consistent between our `Upload` methods.